axelor-account/src/main/java/com/axelor/apps/account/web/BudgetController.java

```diff
@@ -63,4 +64,24 @@ public void checkSharedDates(ActionRequest request, ActionResponse response) {
       response.setError(e.getMessage());
     }
   }
+
+  public void validate(ActionRequest request, ActionResponse response) {
+    Budget budget = request.getContext().asType(Budget.class);
```

Comment:
Also in a controller method, enclose everything in a try catch and trace the exception. Here for example it will trace runtime exception.